smcontroller/smcontroller.go

```diff
@@ -124,48 +143,38 @@ func New(
 		}
 	}
 
+	pb.RegisterSMServiceServer(controller.grpcServer, controller)
+
 	controller.logHandler = map[string]func(cloudprotocol.RequestLog) error{
 		cloudprotocol.SystemLog:  controller.getSystemLog,
 		cloudprotocol.ServiceLog: controller.getServiceLog,
 		cloudprotocol.CrashLog:   controller.getCrashLog,
 	}
 
-	var opts []grpc.ServerOption
+	if err := controller.startGRPCServer(); err != nil {
+		return nil, err
+	}
 
 	if !insecureConn {
-		certURL, keyURL, err := certProvider.GetCertificate(cfg.CertStorage, nil, "")
-		if err != nil {
-			return nil, aoserrors.Wrap(err)
-		}
+		var ch <-chan *iamanager.CertInfo
 
-		tlsConfig, err := cryptcoxontext.GetClientMutualTLSConfig(certURL, keyURL)
-		if err != nil {
+		if ch, err = controller.certProvider.SubscribeCertChanged(cfg.CertStorage); err != nil {
 			return nil, aoserrors.Wrap(err)
 		}
 
-		opts = append(opts, grpc.Creds(credentials.NewTLS(tlsConfig)))
-	} else {
-		log.Info("GRPC server starts in insecure mode")
+		controller.certChan = ch
```

Comment:
You can pass cert channel as parameter instead of using private member.

Reply:
done